tools/keytools/keygen.c

```diff
@@ -1086,8 +1086,45 @@ static void keygen_ml_dsa(const char *priv_fname, uint32_t id_mask)
     fwrite(pub, pub_len, 1, fpriv);
     fclose(fpriv);
 
-    keystore_add(KEYGEN_ML_DSA, pub, pub_len,
-                 priv_fname, id_mask);
+    if (exportPubKey) {
+        if (saveAsDer) {
+            /* Export public key in DER format */
+            uint8_t pubDer[
+            #if ML_DSA_LEVEL == 2
```

Comment:
These come from run-time arguments now. You might need to rebase and fix this.

Reply:
oops. Already had rebased but forgot to change this line. Fixed in latest commit to switch on `ml_dsa_level`